### DISCUSSION TITLE
Add items from packages/ multiple places

### DISCUSSION THREAD

**Discussion by nasirkhan** [spatie/laravel-menu#141] [Q&A]
I installed and added a menu in one of my applications. I add the menu from a Blade Component. I have a number of packages and I want to add items to the same menu if they are installed. 

#### Example Case
I created a `Top Nav` menu as Blade Component. I have a `Blog` package. If I install that package a menu item needs to be displayed in `Top Nav` which I created earlier. 

Is is possible to add items to menus from multiple places? if so, how can I do that?

**Comment by Noaber** [comment]
This would be awesome to have! If anyone has an idea how to accomplish this, that would be great!

**Comment by cristian-araujo** [comment]
Has anyone found a solution? I found two ways to add an item to the menu. The first one adds the menu item to the end of the menu, while the second one adds the menu item to a specific position, but it's not optimized. I'll share the first one.

```PHP
Menu::macro('main', function () {
            $menu = Menu::new()->add() //your menu items.
            Event::dispatch('menu.building', $menu);
            $menu->setActiveFromRequest();
            return $menu;
});
```

in the plugin service provider

```PHP
Event::listen('menu.building', function ($menu) {
          $menu->add() //add new menu items.
});
```
